From c8b823618cf8b9e0510f78a9293887c6b5928a00 Mon Sep 17 00:00:00 2001
Message-Id: <c8b823618cf8b9e0510f78a9293887c6b5928a00.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Fri, 26 Sep 2014 14:49:52 +0530
Subject: [PATCH 10/75] MUST_REBASE [VPG]: drm/i915/adf: ADF wrapper to set
 flag when ADF driver is ready

This is check in interrupt handler later before starting the passthrough
of display related interrupts to ADF driver

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I950f15f8a835b0d161b9e78df5117161c49fba37
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    8 +++++++-
 include/drm/i915_adf.h                  |    7 ++++++-
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 10b1646..058e51d 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -34,6 +34,8 @@
 #include <linux/pm.h>
 #include <linux/pm_runtime.h>
 
+volatile bool g_adf_ready = false;
+
 #ifdef CONFIG_ADF_INTEL
 
 /* Standard MMIO read, non-posted */
@@ -45,7 +47,6 @@
 /* Private register write, double-word addressing, non-posted */
 #define SB_CRWRDA_NP	0x07
 
-
 /* Global for adf driver to get at the current i915 device. */
 static struct drm_i915_private *i915_adf_dev;
 
@@ -59,6 +60,11 @@ void i915_adf_wrapper_teardown(void)
 	i915_adf_dev = NULL;
 }
 
+void set_adf_ready(void)
+{
+	g_adf_ready = true;
+}
+
 struct pci_dev *i915_adf_get_pci_dev(void)
 {
 	if (!i915_adf_dev)
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 2afec81..87f3522 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -36,6 +36,10 @@
 #define INTEL_SIDEBAND_REG_READ		0
 #define INTEL_SIDEBAND_REG_WRITE	1
 
+#ifndef CONFIG_ADF_INTEL
+volatile bool g_adf_ready = false;
+#endif
+
 extern void intel_adf_dpio_sideband_rw(u32 operation, u32 port,
 				       u32 reg, u32 *val);
 extern void intel_adf_pci_sideband_rw(u32 operation, u32 port,
@@ -43,7 +47,8 @@ extern void intel_adf_pci_sideband_rw(u32 operation, u32 port,
 extern struct pci_dev *i915_adf_get_pci_dev(void);
 extern void intel_adf_get_dsi_vbt_data(void **vbt_data,
 				   struct drm_display_mode **mode);
-
+extern void set_adf_ready(void);
+extern volatile bool g_adf_ready;
 #endif
 
 
-- 
1.7.9.5

